app/scripts/modules/core/src/managed/managed.dataSource.ts

```diff
@@ -0,0 +1,44 @@
+import { module, IQService } from 'angular';
+
+import { noop } from 'core/utils';
+import { SETTINGS } from 'core/config/settings';
+import { ApplicationDataSourceRegistry } from 'core/application/service/ApplicationDataSourceRegistry';
+import { Application } from 'core/application';
+import { ManagedReader, IManagedApplicationSummary } from './ManagedReader';
+import {
+  addManagedResourceMetadataToServerGroups,
+  addManagedResourceMetadataToLoadBalancers,
+  addManagedResourceMetadataToSecurityGroups,
+} from './managedResourceDecorators';
+
+export const MANAGED_RESOURCES_DATA_SOURCE = 'spinnaker.core.managed.dataSource';
+module(MANAGED_RESOURCES_DATA_SOURCE, []).run([
+  '$q',
+  ($q: IQService) => {
+    if (!SETTINGS.feature.managedResources) {
+      return;
+    }
+    const loadManagedResources = (application: Application) => {
+      return ManagedReader.getApplicationSummary(application.name);
+    };
+
+    const addManagedResources = (_application: Application, data: IManagedApplicationSummary) => {
+      return $q.when(data);
+    };
+
+    const addManagedMetadataToResources = (application: Application) => {
+      application.serverGroups.ready().then(() => addManagedResourceMetadataToServerGroups(application), noop);
+      application.loadBalancers.ready().then(() => addManagedResourceMetadataToLoadBalancers(application), noop);
+      application.securityGroups.ready().then(() => addManagedResourceMetadataToSecurityGroups(application), noop);
+    };
```

Comment:
I wanted to do something fancy with observables here rather than stealing the way entity tags do it right now (where these functions are also called in the afterLoad handlers of every relevant data source), but it's weird to do today because you have to get an application model in a callback, then bind an observable chain to it, then cleanup that observable chain the next time a callback comes in because you don't have visibility into the application model lifecycle. 

I'm going to follow up with a proposed way of baking cross-data source joins into the data source API itself, since that has awareness of application model lifecycle outside the callbacks exposed here.